geom/type_envelope.go

```diff
@@ -295,3 +295,27 @@ func (e Envelope) box() (rtree.Box, bool) {
 		MaxY: e.maxY,
 	}, !e.IsEmpty()
 }
+
+// BoundingDiagonal returns the LineString that goes from the minimum point in
+// the Envelope to the point in the Envelope. If the envelope is degenerate and
```

Comment:
`from the minimum point in the Envelope to the point in the Envelope`: to which point in the Envelope?

What do you think about describing this as a LineString going from the bottom-left vertex to the top-right vertex of the envelope? At least, that's my understanding of the code, but perhaps my understanding/terminology is too plebeian. 😅

Reply:
Ah, I should have said "from the minimum point in the Envelope to the maximum point in the Envelope".

> What do you think about describing this as a LineString going from the bottom-left vertex to the top-right vertex of the envelope?

That would work if we could assume that geometries were always in a coordinate system where X increases from left to right and Y as you move _up_ the map. WGS84 and the UTM projections would be one such example. Unfortunately it doesn't hold for other projections, such as Web Mertcator, which has Y increasing as you move _down_ the map.

I've reworded the comment in 30468163541b696d898ce5ca5a93e8eb9f249eb3 to explicitly refer to the `Min` and `Max` methods, which is uses to construct the `LineString`.